mantidproject/mantid
# Autocomplete SampleShape in SANS Table

**Description of work.**
Add Hinting tooltip to SampleShape column in SANS v2 table, and parse the input

**Report to:** nobody

**To test:**
1. Interfaces -> SANS -> SANS v2
2. at the bottom, check `Sample Geometry`, which unhides the `SampleShape` column in the table.
3. In the cell in the SampleShape column, enter text which is one of "Disc", "FlatPlate", or "Cylinder". A tooltip should pop up which shows the possible entry values. If you begin to type one of the options, it should autocomplete.
4. Upon entering the data and clicking away from the table, check that text is not changed to an incorrect shape
5. Try removing the input, check that it doesn't revert to your previous input
6. Try some nonsensical text - this should raise an error in the Logger 
7. Load either matching user/batch file combo from the attached data. Batch files can define a sample shape, so check that the sample shape is accepted after loading
8. With a loaded user and batch file, try changing the SampleShape as before. Press `Process All` and check that no errors are thrown

Fixes #23961 
Fixes #24261

[PRdata.zip](https://github.com/mantidproject/mantid/files/2754759/PRdata.zip)

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
